Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -256,5 +290,22 @@ private static void HandleSessionWaitCallback(IntPtr taskCompletionSource, int e
                 tcs.TrySetException(new RealmException(OuterMessage, inner));
             }
         }
+
+        [MonoPInvokeCallback(typeof(NativeMethods.SessionPropertyChangedCallback))]
+        private static void HandleSessionPropertyChangedCallback(IntPtr managedSessionHandle, NotifiableProperty property)
+        {
+            var propertyName = property switch
+            {
+                NotifiableProperty.ConnectionState => nameof(Session.ConnectionState),
+                _ => throw new NotSupportedException($"Unexpected notifiable property value: {property}")
+            };
+            var session = (Session)GCHandle.FromIntPtr(managedSessionHandle).Target;
+            session.RaisePropertyChanged(propertyName);
```

Comment:
```suggestion
            System.Threading.ThreadPool.QueueUserWorkItem(_ =>
            {
                session.RaisePropertyChanged(propertyName);
            });
```
After what we learned in #2830 I think we should always invoke all user callbacks on the thread pool instead of directly on the sync client's worker thread.